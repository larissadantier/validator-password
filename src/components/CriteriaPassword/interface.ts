import { ComponentProps } from "react";

export interface ITextProps extends ComponentProps<'span'> {
  valid?: boolean
}

export interface ITitleProps extends ComponentProps<'h4'> {
  valid?: boolean
}