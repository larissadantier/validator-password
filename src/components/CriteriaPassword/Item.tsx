import { ComponentProps } from "react";

function Item({ children, ...props }: ComponentProps<'li'>) {
  return (
    <li data-testid="validator__item" {...props}>{children}</li>
  );
}

export default Item;