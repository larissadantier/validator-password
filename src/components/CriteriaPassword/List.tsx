import { ComponentProps } from "react";

import styles from './validator-password.module.scss';

function List({ children, ...props }: ComponentProps<'ul'>) {
  return (
    <ul data-testid="validator__list" className={styles['validator__list']} {...props}>{children}</ul>
  );
}

export default List;