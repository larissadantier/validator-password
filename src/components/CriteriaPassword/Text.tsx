import { ITextProps } from "./interface";

import styles from './validator-password.module.scss';

function Text({ children, valid, ...props }: ITextProps) {
  const classNameValided = valid
    ? `${styles.validator} ${styles['validator--valid']}`
    : `${styles.validator} ${styles['validator--invalid']}`

  return (
    <span data-testid="validator__text" className={classNameValided} {...props}>
      - {children}
    </span>
  );
}

export default Text;