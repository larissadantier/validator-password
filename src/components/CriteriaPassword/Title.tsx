import { ITitleProps } from "./interface";

import styles from './validator-password.module.scss';

function Title({ children, valid, ...props }: ITitleProps) {
  const classNameValided = valid
    ? `${styles.validator} ${styles['validator--valid']}`
    : `${styles.validator} ${styles['validator--invalid']}`

  return (
    <h4 data-testid="validator__title" className={classNameValided} {...props}>{children}</h4>
  );
}

export default Title;