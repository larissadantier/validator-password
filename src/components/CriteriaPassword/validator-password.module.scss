.validator {
  font-size: 0.875rem;

  &__list {
    display: flex;
    flex-direction: column;

    gap: 0.25rem;

    &>li {
      margin-left: 1rem;
    }
  }

  &--valid {
    color: var(--success);
  }

  &--invalid {
    color: var(--error);
  }
}