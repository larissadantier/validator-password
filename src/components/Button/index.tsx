import Root from "./Root";
import Text from './Text';

export const Button = {
  Root,
  Text
}