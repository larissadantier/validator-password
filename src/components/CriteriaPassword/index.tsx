import List from './List';
import Item from './Item';
import Title from './Title';
import Text from './Text';

export const CriteriaPassword = {
  List,
  Item,
  Title,
  Text
}