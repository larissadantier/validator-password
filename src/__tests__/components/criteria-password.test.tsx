import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

import { CriteriaPassword } from "../../components/CriteriaPassword";

const criteriaPasswordListTestId = 'validator__list';
const criteriaPasswordItemTestId = 'validator__item';
const criteriaPasswordTitleTestId = 'validator__title';
const criteriaPasswordTextTestId = 'validator__text';

describe('Component CriteriaPassword', () => {
  it('should be able render the list', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>
    )

    expect(getByTestId(criteriaPasswordListTestId)).toBeInTheDocument()
  })

  it('should be able render title', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Title>Titulo</CriteriaPassword.Title>

        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>
    )

    expect(getByTestId(criteriaPasswordTitleTestId)).toHaveTextContent('Titulo')
  })

  it('should be able render the item', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>
    )

    expect(getByTestId(criteriaPasswordItemTestId)).toHaveTextContent('Item 1')
  })

  it('should be able text have a valid item', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Item>
          <CriteriaPassword.Text valid>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>)

    expect(getByTestId(criteriaPasswordTextTestId).className).toMatch(/validator--valid/);
  })

  it('should be able text have a not valid item', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>)

    expect(getByTestId(criteriaPasswordTextTestId).className).toMatch(/validator--invalid/);
  })

  it('should be able title have a valid', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Title>Titulo</CriteriaPassword.Title>

        <CriteriaPassword.Item>
          <CriteriaPassword.Text valid>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>)

    expect(getByTestId(criteriaPasswordTextTestId).className).toMatch(/validator--valid/);
  })

  it('should be able title have a not valid', () => {
    const { getByTestId } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Title>Titulo</CriteriaPassword.Title>

        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>)

    expect(getByTestId(criteriaPasswordTextTestId).className).toMatch(/validator--invalid/);
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <CriteriaPassword.List>
        <CriteriaPassword.Item>
          <CriteriaPassword.Text>Item 1</CriteriaPassword.Text>
        </CriteriaPassword.Item>
      </CriteriaPassword.List>
    );

    expect(asFragment()).toMatchSnapshot();
  });
})