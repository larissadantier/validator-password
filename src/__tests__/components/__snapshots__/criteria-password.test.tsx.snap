// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component CriteriaPassword > should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="_validator__list_52daaf"
    data-testid="validator__list"
  >
    <li
      data-testid="validator__item"
    >
      <span
        class="_validator_52daaf _validator--invalid_52daaf"
        data-testid="validator__text"
      >
        - Item 1
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
